File: yyong/2025.05/250505_마라톤_1.py
N = int(input())
pos = [list(map(int, input().split())) for _ in range(N)]

cur = pos[0]
total_dist = 0

# 총 거리 및 위치 구하기
for i in range(1, N):
    x, y = pos[i]
    total_dist += abs(cur[0] - x) + abs(cur[1] - y)
    cur = [x, y]

result = float('inf')

# 하나씩 없애보기
for j in range(1, N - 1):
    a = abs(pos[j - 1][0] - pos[j][0]) + abs(pos[j - 1][1] - pos[j][1])
    b = abs(pos[j + 1][0] - pos[j][0]) + abs(pos[j + 1][1] - pos[j][1])
    c = abs(pos[j + 1][0] - pos[j - 1][0]) + abs(pos[j + 1][1] - pos[j - 1][1])

    new_dist = total_dist - (a + b) + c
    result = min(result, new_dist)

print(result)

File: yyong/2025.05/250505_마라톤_1.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const N = Number(input[0]);
const pos = [];

let cur = input[1].split(" ").map(Number);
pos.push(cur);

let totalDist = 0;

// 총 거리 및 위치 구하기
for (let i = 2; i < N + 1; i++) {
  const [x, y] = input[i].split(" ").map(Number);
  pos.push([x, y]);
  totalDist += Math.abs(cur[0] - x) + Math.abs(cur[1] - y);
  cur = [x, y];
}

let result = Number(Infinity);

// 하나씩 없애보기
for (let j = 1; j < N - 1; j++) {
  const a =
    Math.abs(pos[j - 1][0] - pos[j][0]) + Math.abs(pos[j - 1][1] - pos[j][1]);
  const b =
    Math.abs(pos[j + 1][0] - pos[j][0]) + Math.abs(pos[j + 1][1] - pos[j][1]);
  const c =
    Math.abs(pos[j + 1][0] - pos[j - 1][0]) +
    Math.abs(pos[j + 1][1] - pos[j - 1][1]);

  const newDist = totalDist - (a + b) + c;

  result = Math.min(result, newDist);
}

console.log(result);
